src/services/platform/platformTransformers.js

```diff
@@ -70,11 +70,13 @@ const userPermissions = (response, { config = rbacConfig } = {}) => {
         updatedResponse.permissions[app].all = true;
       }
 
-      if (!updatedResponse.permissions[app].resources[resource]) {
+      if (resource && !updatedResponse.permissions?.[app]?.resources?.[resource]) {
```

Comment:
required: this can be simplified into a nullish assignment `??= {}`

```
if (resource) {
  updatedResponse.permissions[app].resources[resource] ??= {};

  if (operation) {
    updatedResponse.permissions[app].resources[resource][operation] = definitions;
  }
}
```